### DISCUSSION TITLE
How to overload comparison operators in the extensions?

### DISCUSSION THREAD

**Discussion by dash-awake** [duckdb/duckdb#7473] [Q&A]
We are adding a few operators to the inet extension. Using the example of the `-` operator in the existing code, we added a `<` operator as below:

```
    auto lessthan_fun = ScalarFunction("<", {inet_type, inet_type}, LogicalType::BOOLEAN, INetFunctions::LessThan);
    CreateScalarFunctionInfo lessthan_info(lessthan_fun);
    lessthan_info.on_conflict = OnCreateConflict::ALTER_ON_CONFLICT;
    catalog.CreateFunction(*con.context, &lessthan_info);
```
But, the following expression doesn't invoke the `LessThan` function.

```
SELECT INET '192.168.1.253/23' < INET '192.168.1.250/24';
```
New operators that postgres supports -- such as `<<=` work fine. Seems like the comparison operators are special - `<`, `>`, `=`.

Is there a special way to override the comparison operators?

**Comment by Mytherin** [comment]
There is currently no support yet for overloading comparison operators - they are all based on the underlying physical type. In the future we would like to add support for this but currently this is not possible from within an extension.
